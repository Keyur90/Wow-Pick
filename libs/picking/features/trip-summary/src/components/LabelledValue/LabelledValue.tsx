import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

interface LabelledValuePropTypes {
  label: string;
  value: string | number | React.ReactNode;
}

const useStyles = makeStyles((theme) => ({
  label: {
    marginRight: theme.spacing(0.5),
  },
  value: {
    fontWeight: 500,
    color: theme.palette.common.black,
  },
}));

const LabelledValue: React.FC<LabelledValuePropTypes> = React.memo(({ label, value }) => {
  const classes = useStyles();

  return (
    <span>
      <span className={classes.label}>{label}:</span>
      <span className={classes.value}>{value}</span>
    </span>
  );
});

export { LabelledValue };
