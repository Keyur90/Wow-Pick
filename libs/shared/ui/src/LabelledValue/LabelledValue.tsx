import { makeStyles, Theme } from '@material-ui/core/styles';
import React from 'react';

interface LabelledValuePropTypes {
  label?: string;
  value: string | number | React.ReactNode;
  emphasis?: boolean;
}

interface StyledProps {
  emphasis: boolean;
}

const useStyles = makeStyles<Theme, StyledProps>((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
  },
  label: {
    fontWeight: 500,
    fontSize: '0.75rem',
    lineHeight: '0.875rem',
    color: ({ emphasis }) => (emphasis ? '#171C1F' : '#899195'),
    letterSpacing: '0.75px',
  },
  value: {
    fontWeight: 500,
    fontSize: '1.25rem',
    lineHeight: '1.5rem',
    color: '#171C1F',
  },
}));

const LabelledValue: React.FC<LabelledValuePropTypes> = ({ label, value, emphasis = false }) => {
  const classes = useStyles({ emphasis });

  return (
    <div className={classes.root}>
      {label && <div className={classes.label}>{label}</div>}
      <div className={classes.value}>{value}</div>
    </div>
  );
};

export { LabelledValue };
