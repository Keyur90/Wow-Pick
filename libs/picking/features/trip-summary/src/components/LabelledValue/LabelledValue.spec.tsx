import React from 'react';
import { render } from '@testing-library/react';
import { LabelledValue } from './LabelledValue';

test('renders LabelledValue', () => {
  const { container } = render(<LabelledValue label={'Label'} value={'Value'} />);

  expect(container).toMatchSnapshot();
});
