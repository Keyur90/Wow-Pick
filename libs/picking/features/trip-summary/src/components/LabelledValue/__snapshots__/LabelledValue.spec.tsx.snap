// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LabelledValue 1`] = `
<div>
  <span>
    <span
      class="makeStyles-label-1"
    >
      Label
      :
    </span>
    <span
      class="makeStyles-value-2"
    >
      Value
    </span>
  </span>
</div>
`;
